Classes/Task/OptimizeIndexTaskAdditionalFieldProvider.php

```diff
@@ -0,0 +1,281 @@
+ */
+class OptimizeIndexTaskAdditionalFieldProvider extends AbstractAdditionalFieldProvider
+{
+    /**
+     * Default language file of the extension linkvalidator
+     *
+     * @var string
+     */
+    protected $languageFile = 'LLL:EXT:solr/Resources/Private/Language/locallang.xlf';
+
+    /**
+     * Task information
+     *
+     * @var array
+     */
+    protected $taskInformation;
+
+    /**
+     * Scheduler task
+     *
+     * @var AbstractTask|ReIndexTask|NULL
+     */
+    protected $task = null;
+
+    /**
+     * Scheduler Module
+     *
+     * @var SchedulerModuleController
+     */
+    protected $schedulerModule;
+
+    /**
+     * Selected site
+     *
+     * @var Site
+     */
+    protected $site = null;
+
+    /**
+     * SiteRepository
+     *
+     * @var SiteRepository
+     */
+    protected $siteRepository;
+
+    /**
+     * @var PageRenderer
+     */
+    protected $pageRenderer = null;
+
+    /**
+     * ReIndexTaskAdditionalFieldProvider constructor.
+     */
+    public function __construct()
+    {
+        $this->siteRepository = GeneralUtility::makeInstance(SiteRepository::class);
+    }
+
+    /**
+     * @param array $taskInfo
+     * @param \TYPO3\CMS\Scheduler\Task\AbstractTask|NULL $task
+     * @param \TYPO3\CMS\Scheduler\Controller\SchedulerModuleController $schedulerModule
+     */
+    protected function initialize(
+        array $taskInfo = [],
+        AbstractTask $task = null,
+        SchedulerModuleController $schedulerModule
+    )
+    {
+        /** @var $task ReIndexTask */
+        $this->taskInformation = $taskInfo;
+        $this->task = $task;
+        $this->schedulerModule = $schedulerModule;
+
+        $currentAction = $schedulerModule->getCurrentAction();
+
+        if ($currentAction->equals(Action::EDIT)) {
+            $this->site = $this->siteRepository->getSiteByRootPageId($task->getRootPageId());
+        }
+    }
+
+    /**
+     * Used to define fields to provide the Solr server address when adding
+     * or editing a task.
+     *
+     * @param array $taskInfo reference to the array containing the info used in the add/edit form
+     * @param AbstractTask $task when editing, reference to the current task object. Null when adding.
+     * @param SchedulerModuleController $schedulerModule reference to the calling object (Scheduler's BE module)
+     * @return array Array containing all the information pertaining to the additional fields
+     *                        The array is multidimensional, keyed to the task class name and each field's id
+     *                        For each field it provides an associative sub-array with the following:
+     */
+    public function getAdditionalFields(
+        array &$taskInfo,
+        $task,
+        SchedulerModuleController $schedulerModule
+    )
+    {
+        $additionalFields = [];
+
+        if (!$this->isTaskInstanceofOptimizeIndexTask($task)) {
+            return $additionalFields;
+        }
```

Comment:
Method `isTaskInstanceofOptimizeIndexTask()` never returns a false value, instead an exception is thrown.
The return statement never reached.

Reply:
Was adapted from `Classes/Task/ReIndexTaskAdditionalFieldProvider.php` and not really cared about by me. I change it in my new file.

Reply:
That was just fine, it must return true only or throw the error to avoid the selector in other classes.